### PR TITLE
fix(responses): preserve function calls completed in done events

### PR DESCRIPTION
**Pull request by MrLiuGangQiang**
## Summary

Fixes an OpenAI Responses streaming conversion issue where a function call completed by `response.function_call_arguments.done` can lose its arguments or tool identity before it reaches the client.

This affects upstreams that provide arguments, `name`, `namespace`, or `call_id` only in a later event. In Codex, that can leave `collaboration.spawn_agent` without a dispatchable identity even when the downstream response is valid.

## Root cause

The outbound transformer previously treated `response.function_call_arguments.done` as state-only. The inbound transformer also emitted `response.output_item.added` immediately, even when the function name was absent. A later terminal event could repair the final item, but it could not repair the already-emitted initial item that clients use to route the call.

### Outbound Responses transformer

- Resolves calls from `call_id`, with an `item_id` to `call_id` fallback.
- Treats non-empty terminal arguments as authoritative.
- Emits the missing argument suffix after partial deltas, without duplicating complete payloads.
- Emits an identity-only unified delta when `name` or `namespace` arrives after all arguments have already streamed.
- Preserves `id`, `type`, `name`, and `namespace` in emitted tool-call deltas.
- Returns an explicit mismatch error for divergent terminal arguments.

### Inbound Responses transformer

- Defers `output_item.added` for unnamed function calls until a tool name is available.
- Buffers arguments received before tool identity and replays them only after emitting the correctly identified `output_item.added`.
- Merges late tool identity into tracked state.
- Includes the final `call_id`, `name`, `namespace`, and arguments in terminal `response.function_call_arguments.done` and `response.output_item.done` events.

## JSON comparison safety

Semantically equivalent JSON that differs only in whitespace or object-key order is accepted without a duplicate delta. Comparisons use `json.Decoder.UseNumber()`, preserving numeric lexemes so distinct large integers cannot collapse through `float64` conversion.

## Regression coverage

- Complete arguments provided only in `function_call_arguments.done`.
- Reformatted but equivalent terminal JSON.
- Distinct large integers (`9007199254740992` and `9007199254740993`) rejected as a mismatch.
- Complete outbound to unified to inbound round trip when terminal events provide `spawn_agent`, `collaboration`, call ID, and JSON arguments.
- Arguments that arrive before late tool identity are buffered until a correctly identified `output_item.added` is emitted.

## Verification

```text
cd llm
go test ./transformer/openai/responses -run 'TestOutboundTransformer_TransformStream_(EmitsArgumentsProvidedOnlyInDone|AcceptsEquivalentFinalArguments|RejectsDistinctLargeJSONNumbers)|TestResponsesStream_RoundTrip_(PreservesToolIdentityProvidedOnlyInDone|DefersFunctionCallUntilLateIdentityArrives)' -count=1
git diff --check
```

### CODE REVIEW

File: llm/transformer/openai/responses/outbound_stream.go [hunk 1]
```diff
@@ -332,20 +332,67 @@ func (s *responsesOutboundStream) transformStreamChunk(event *httpclient.StreamE
...
+
+		forwardedArgs := tc.Function.Arguments
+		var missingArgs string
+		switch {
+		case forwardedArgs == "":
+			missingArgs = finalArgs
+		case strings.HasPrefix(finalArgs, forwardedArgs):
+			missingArgs = strings.TrimPrefix(finalArgs, forwardedArgs)
+		default:
+			return fmt.Errorf("function call arguments mismatch for call_id %q", callID)
```
[the remark this answers is not in the document]
Comment (MrLiuGangQiang) [reply llm/transformer/openai/responses/outbound_stream.go]: Addressed in 3faf38df.

The reconciliation now accepts complete, semantically equivalent JSON values even when the final payload differs only by whitespace or object-key order. In that case the already-forwarded arguments remain the sole delta, while the final payload is retained as canonical state for terminal events. Truly divergent values still fail explicitly to avoid emitting corrupted concatenated arguments.

Added a regression test covering reordered keys and whitespace differences.

[the remark this answers is not in the document]
Comment (MrLiuGangQiang) [reply llm/transformer/openai/responses/outbound_stream.go]: Fixed in 98e5a36d. The semantic JSON comparison now uses `json.Decoder.UseNumber()` before `reflect.DeepEqual`, preserving numeric lexemes rather than converting numbers to `float64`. This prevents distinct values such as `9007199254740992` and `9007199254740993` from comparing equal and suppressing the authoritative `arguments.done` payload. I also added a regression test covering that exact case; the existing whitespace/key-order equivalence test remains covered.

**Comment by MrLiuGangQiang** [comment]
主要是为了解决codex使用子agent报错的问题
